File: app/Models/Attachment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Attachment extends Model
{
    use HasFactory;

    protected $fillable = [
        'file_path',
        'file_name',
        'file_mime_type',
        'file_size',
        'uploaded_by',
    ];

    protected $appends = ['file_url'];

    public function getFileUrlAttribute()
    {
        return rtrim(config('app.url'), '/') . '/storage/attachments/' . basename($this->file_path);
    }

    /**
     * Get the parent attachable model (task or roadmap).
     */
    public function attachable()
    {
        return $this->morphTo();
    }

    /**
     * Get the user who uploaded the attachment.
     */
    public function uploader()
    {
        return $this->belongsTo(User::class, 'uploaded_by');
    }
}

File: mobile_app/lib/models/attachment.dart
class Attachment {
  final int id;
  final String attachableType;
  final int attachableId;
  final String filePath;
  final String fileName;
  final String fileMimeType;
  final int fileSize;
  final String? fileUrl;

  Attachment({
    required this.id,
    required this.attachableType,
    required this.attachableId,
    required this.filePath,
    required this.fileName,
    required this.fileMimeType,
    required this.fileSize,
    this.fileUrl,
  });

  factory Attachment.fromJson(Map<String, dynamic> json) {
    return Attachment(
      id: json['id'],
      attachableType: json['attachable_type'],
      attachableId: json['attachable_id'],
      filePath: json['file_path'],
      fileName: json['file_name'],
      fileMimeType: json['file_mime_type'],
      fileSize: json['file_size'],
      fileUrl: json['file_url'],
    );
  }
}
